src/Resource/CloudAccount/Endpoint.php

```diff
@@ -148,6 +151,139 @@ protected function _waitUntilPhpVersion(
     };
   }
 
+  /**
+   * Create a backup
+   *
+   * @param Entity An instance of cloud account entity.
+   * @return Backup
+   * @throws ApiException If request fails
+   */
+  public function createBackup(Entity $entity) : Backup {
+    $this->wait(null);
+    $response = $this->_client->request(
+      'POST',
+      self::_URI . "/{$entity->getId()}/backup"
+    );
+
+    return $this->getModel(Backup::class)->sync($response);
+  }
+
+  /**
+   * Return a list of backups
+   *
+   * @return Backup
+   * @throws ApiException If request fails
+   */
+  public function getBackups(Entity $entity) : Collection {
+    $this->wait(null);
+    $collection = new Collection(Backup::class);
+    
+    foreach ($this->_fetchBackupList($entity) as $backup) {
+      $collection->add($this->getModel(Backup::class)->sync($backup));
+    }
+
+    return $collection;
+  }
+
+  /**
+   * Return a specific backup
+   *
+   * @param string $file_name The unique file name for the backup to retrieve.
+   * @return Backup
+   * @throws ApiException If request fails
+   */
+  public function getBackup(Entity $entity, string $file_name) : Backup {
+    $this->wait(null);
+    return $this->_findBackup($entity, $file_name);
+  }
+
+  /**
+   * Download a specific backup
+   *
+   * @param string $file_name The unique file name for the backup to retrieve.
+   * @param string $path the directory to store the download in.
+   *
+   * @return Promise
+   * @throws ApiException If request fails
+   * @throws Exception
+   */
+  public function downloadBackup(Entity $entity, string $file_name, string $path) {
+    $this->wait(null);
+
+    if (! file_exists($path) || ! is_dir($path)) {
+      throw new Exception('##LG_INVALID_PATH##');
+    }
+
+    $path = trim($path);
+    if (substr($path, -1) !== DIRECTORY_SEPARATOR) {
+      $path .= DIRECTORY_SEPARATOR;
+    }
+
+    $save_to = $path . $file_name;
+
+    if (file_exists($save_to)) {
+      throw new Exception('##LG_FILE_ALREADY_EXISTS##');
+    }
+
+    $stream = fopen($save_to,'w');
+
+    $this->_client->request(
```

Comment:
this is behaving weird.  if i pass a valid $path and an invalid $file_name, I get an exception that the backup was not found, but I also get an empty file at `$path/$file_name`.
```php
$config = new NexcessConfig([
  'api_token' => '...',
  'base_uri' => 'https://demo2.nocworx.com/extranet/',
  'debug' => true
]);

$api = new Client($config);

$cloudAccounts = $api->getEndpoint(Endpoint::class);
$cloud = $cloudAccounts->retrieve(1642);

$cloudAccounts->downloadBackup($cloud, 'nosuchbackup', '.');

// got your '##LG_BACKUP_NOT_FOUND##' exception
// there's now an empty file named  ./nosuchbackup
```

**edit** ah, figured it out: need to `_findBackup()` before doing `fopen()`.

Reply:
I put the whole thing in a try catch and in the catch I do cleanup.